### ISSUE TITLE
Running 'yarn build' fails with TypeScript error

### ISSUE THREAD

**Issue by 10thfloor** [fission-suite/keystore-idb#28]
# Summary
Running `yarn build` fails.

MacOS: `10.15.5`
Node: `12.18.1`
Typescript: `2.7.9`

WIth the output: 

```
src/ecc/keys.ts:27:10 - error TS2769: No overload matches this call.
  Overload 1 of 3, '(format: "jwk", keyData: JsonWebKey, algorithm: string | Algorithm | AesKeyAlgorithm | EcKeyImportParams | HmacImportParams | RsaHashedImportParams | DhImportKeyParams, extractable: boolean, keyUsages: KeyUsage[]): PromiseLike<...>', gave the following error.
    Argument of type '"raw"' is not assignable to parameter of type '"jwk"'.
  Overload 2 of 3, '(format: "raw" | "pkcs8" | "spki", keyData: ArrayBuffer | DataView | Int8Array | Uint8Array | Uint8ClampedArray | ... 5 more ... | Float64Array, algorithm: string | ... 5 more ... | DhImportKeyParams, extractable: boolean, keyUsages: KeyUsage[]): PromiseLike<...>', gave the following error.
    Argument of type 'string[]' is not assignable to parameter of type 'KeyUsage[]'.
  Overload 3 of 3, '(format: string, keyData: ArrayBuffer | DataView | Int8Array | Uint8Array | Uint8ClampedArray | Int16Array | ... 5 more ... | JsonWebKey, algorithm: string | ... 5 more ... | DhImportKeyParams, extractable: boolean, keyUsages: KeyUsage[]): PromiseLike<...>', gave the following error.
    Argument of type 'string[]' is not assignable to parameter of type 'KeyUsage[]'.

 27   return window.crypto.subtle.importKey(
             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 28     'raw',
    ~~~~~~~~~~
... 
 32     uses
    ~~~~~~~~
 33   )
    ~~~
```

**Comment by bmann** [comment]
Thanks @10thfloor -- we'll take a look on Monday.

**Comment by bmann** [comment]
I just tried this and it worked for me with:

MacOS: `10.15.5` (Catalina)
Node: `14.2.0`
yarn: `1.22.4`

I didn't have typescript installed on my system, I installed it globally from npm:

TypeScript: `3.9.7`

I'm pretty sure a newer version of node is required -- and TypeScript -- so we should update package.json. Thanks for reporting!

**Comment by dholms** [comment]
I don't think that this is a node issue, because it build fine as far back as node v10 (https://travis-ci.org/github/fission-suite/keystore-idb/builds/709731294)

I was able to replicate the issue by deleting `yarn.lock` and reinstalling. I believe that Typescript updated it's internal `SubtleCrypto` interface, which is causing the build to break.

@10thfloor are you installing dependencies with `yarn` or `npm install`? `yarn` should use our lock file and thus the same version of the compiler. That being said, we should fix this to match the latest version of the TSC. I'll take a look at that shortly

Thanks for the issue report!

**Comment by 10thfloor** [comment]
I used Yarn to install. I was able to workaround the issue by downgrading TypeScript to 3.8.2 (version specified in this repos `package.json`). It seems the problem for me was the `lib.dom.d.ts` type definitions for `subtle.crypto` are incorrect when running newest TypeScript. Not sure if this is user error or not 😓 ...

**Comment by dholms** [comment]
Yup typescript updated their interface for `SubtleCrypto`, using a new `KeyUsage` type instead of `string`s
I merged a fix here: https://github.com/fission-suite/keystore-idb/pull/32
and published as `keystore-idb@0.13.2`

Give it a go and let me know if it works for you @10thfloor

**Comment by 10thfloor** [comment]
@dholms It works! ty.
